api/app/Http/Controllers/MediaController.php

```diff
@@ -186,82 +217,282 @@ public function showOneMedia($id){
 	 *		 required=false,
 	 *		 @OA\Schema(type="string", maxLength=150)
 	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="hashFunction",
+	 *		 in="query",
+	 *		 description="Cryptographic hash function used to compute the value given in the Hash Value",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=45)
+	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="hashValue",
+	 *		 in="query",
+	 *		 description="Value computed by a hash function applied to the media that will be delivered at the access point",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=45)
+	 *	 ),
 	 *	 @OA\Response(
-	 *		 response="200",
+	 *		 response="201",
 	 *		 description="Returns UUID recordID (GUID)",
 	 *		 @OA\JsonContent()
 	 *	 ),
 	 *	 @OA\Response(
 	 *		 response="400",
 	 *		 description="Error: Bad request.",
 	 *	 ),
+	 *	 @OA\Response(
+	 *		 response="401",
+	 *		 description="Unauthorized",
+	 *	 ),
 	 * )
 	 */
-	public function insertOccurrenceMedia(Request $request){
-		if($user = $this->authenicate($request)){
-			$this->validate($request, [
-				'occid' => 'integer',
-				'originalUrl' => 'required'
-				'format' => '',
-				'originalUrl
-				'mediumUrl
-				'thumbnailUrl
-				'archiveUrl
-				'referenceUrl
-				'occid
-				'tid
-				'photographer
-				'photographerUid
-				'imageType
-				'caption
-				'copyright
-				'rights
-				'accessRights
-				'locality
-				'notes
-				'sourceIdentifier
-			]);
-			$inputArr = $request->all();
-			$occid = $request->input('occid');
-			if(!$occid){
-				if($guid = $request->input('guid')){
-					$occid = Occurrence::where('occurrenceID', $guid)->value('occid');
-					if(!$occid) $occid = DB::table('guidoccurrences')->where('guid', $guid)->value('occid');
-				}
-				if(!$occid){
-					if($catalogNumber = $request->input('catalogNumber')){
-						$occid = Occurrence::where('catalogNumber', $catalogNumber)->value('occid');
-					}
-				}
-				if($occid) $inputArr['occid'] = $occid;
-			}
-			if($occid){
-				//Media object is to be linked to an occurrence
-				//Check if user is authorized to modify target collection
-				Occurrence::where()->value('');
-
-				$media = Media::create($inputArr);
-			}
-			else{
-				return response()->json(['error' => 'Unauthorized'], 401);
-			}
+	public function insert(Request $request){
+		if($user = $this->authenticate($request)){
+			$request->validate($this->rulesInsert);
 
+			$inputArr = $request->all();
+			$this->adjustInputData($inputArr);
+			$media = Media::create($inputArr);
```

Comment:
If any of these steps fail, it looks like a 201 response will still be returned. Do we need to nest this in in a try catch and/or verify the success of some of these steps?